### PR TITLE
Limit Yoga layout to relevant root on resize

### PR DESCRIPTION
**Pull request by rozele**
### Type of Change
_Erase all that don't apply._
- Perf (non-breaking change which improves performance)

### Why
Recalculation of Yoga layout due to root view size change does not require that we re-run layout on all roots. It also does not require that we invoke UpdateLayout on any new nodes (as a resize operation is independent of a batch of UIManager operations).

### What
This change limits the scope of Yoga calculation and application of Yoga results to XAML views to only the root view that changed size. This optimization primarily impacts apps with multiple windows, but can also benefit any React Native Windows app with multiple roots.

In a future PR, we could also track impacted roots in view manager operations by looking up the rootTag field on the ShadowNode, and only running Yoga layout on windows with actual updates in a given batch.

## Testing
Ran RNTester, window resize still adapts correctly.

https://user-images.githubusercontent.com/1106239/190245420-555c1b14-471a-4866-b04d-cd0fc0aebcde.mp4

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10580)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Views/ViewManagerBase.cpp [hunk 1]
```diff
@@ -319,6 +319,7 @@ void ViewManagerBase::NotifyUnimplementedProperty(
     ShadowNodeBase *nodeToUpdate,
     const std::string &propertyName,
     const winrt::Microsoft::ReactNative::JSValue &value) {
+#ifndef ARCHONRNW
```
Comment (acoates-ms) [inline vnext/Microsoft.ReactNative/Views/ViewManagerBase.cpp:322]: This looks unrelated?
Separately we should probably look at disabling this by default --  it does produce a lot of debug spew which most apps are not going to want clogging up their debug output.

Comment (rozele) [reply vnext/Microsoft.ReactNative/Views/ViewManagerBase.cpp:322]: Yeah, my bad. Was testing if disabling those logs would make the debug build any faster and this leaked in 😬

Comment (rozele) [reply vnext/Microsoft.ReactNative/Views/ViewManagerBase.cpp:322]: Fixed :)

File: vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp [hunk 2]
```diff
@@ -217,23 +217,26 @@ void NativeUIManager::AddRootView(ShadowNode &shadowNode, facebook::react::IReac
...
 
   auto element = view.as<xaml::FrameworkElement>();
-  Microsoft::ReactNative::SetTag(element, shadowNode.m_tag);
+  Microsoft::ReactNative::SetTag(element, rootTag);
 
   // Add listener to size change so we can redo the layout when that happens
-  m_sizeChangedVector.push_back(
-      view.as<xaml::FrameworkElement>().SizeChanged(winrt::auto_revoke, [this](auto &&, auto &&) { DoLayout(); }));
+  m_sizeChangedVector.push_back(view.as<xaml::FrameworkElement>().SizeChanged(
+      winrt::auto_revoke, [this, rootTag](auto &&, xaml::SizeChangedEventArgs const &args) {
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp:237]: Is it safe to capture `rootTag`, or should it be a weak_ref? Is there a scenario where this thing gets torn down and we need to care about this SizeChanged event firing? Probably not?

I noticed that `m_sizeChangedVector` doesn't have the element removed in `removeRootView`. Preexisting, but... given that you're capturing a value now it made me wonder if that's all buttoned up.

File: vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp [hunk 3]
```diff
@@ -217,23 +217,26 @@ void NativeUIManager::AddRootView(ShadowNode &shadowNode, facebook::react::IReac
...
+        ApplyLayout(rootTag, args.NewSize().Width, args.NewSize().Height);
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp:238]: This is slightly different than what was there before. This takes the size right from the size changed event. The previous implementation (via `DoLayout`) would use the `ActualSize` of the XAML type. Are those for sure the same? Sounds like... [sure](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.frameworkelement.sizechanged?view=winrt-22621)? In which case this is fine. Just checking.

**Review by chrisglein — APPROVED** [review]
